### DISCUSSION TITLE
Conditions in MDF

### DISCUSSION THREAD

**Discussion by Monsurat-Onabajo** [ModECI/MDF#382] [Q&A]
Hello,

I was trying to create a condition in MDF using the Convenience condition  'AtEnvironmentStateUpdateStart'  it ran when i evaluated it but when i checked the results of each nodes, i kept on getting None. When i changed the condition to 'AtEnvironmentStateUpdateNStart', it ran for hours and it didn't stop running until i canceled it. Is it that MDF doesn't support the convenience condition category or i am missing something?

**Comment by pgleeson** [comment]
@Monsurat-Onabajo Can you paste the relevant sections of the code here? 

@kmantel Any thoughts on this?

**Comment by Monsurat-Onabajo** [comment]
Thank you @pgleeson This is the condition i was trying to use in the graph i created And below it is the output i got, i expected numbers but i was getting 'None'
![IMG_20230120_175931](https://user-images.githubusercontent.com/102332122/213759483-fd9c26cc-9dae-4d8f-99ec-4fb32dd6718a.jpg)
![IMG_20230120_180007](https://user-images.githubusercontent.com/102332122/213759962-14107659-e1eb-499b-8ddd-3a430d439608.jpg)

**Comment by kmantel** [comment]
It's a bit difficult to tell without the model and code, but using [AtEnvironmentStateUpdateStart](https://kmantel.github.io/graph-scheduler/Condition.html#graph_scheduler.condition.AtEnvironmentStateUpdateStart) as a termination condition for EnvironmentStateUpdate should terminate immediately, before running anything, which could explain the None values. As for AtEnvironmentStateUpdateNStart not terminating when also used as the EnvironmentStateUpdate termination condition, a hint could be the fact that a call to scheduler.run is the scope of a EnvironmentStateUpdate. This means that unless a batch of runs are being called, the value of the current EnvironmentStateUpdate should be 0, in which case any non-default value for n (for n != 0), this condition may never be reached.

I think these sections of the scheduler docs might be helpful here, especially Time: 

https://kmantel.github.io/graph-scheduler/Time.html
https://kmantel.github.io/graph-scheduler/Condition.html#execution

if it's still not clear, feedback on the docs is of course welcome, but looking at some test examples using these conditions might also be helpful: https://github.com/kmantel/graph-scheduler/blob/2f4d8764b7e0311e85c405515ab7d2bb9255f6a0/tests/scheduling/test_condition.py#L576

**Reply by Monsurat-Onabajo** [reply]
Thank you very much, i will take a look at them and revert back to you if i am stuck again
